languagetool-language-modules/ga/src/main/java/org/languagetool/rules/ga/DhaNoBeirtRule.java

```diff
@@ -0,0 +1,97 @@
+/* LanguageTool, a natural language style checker
+ * Copyright (C) 2005 Daniel Naber (http://www.danielnaber.de)
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301
+ * USA
+ */
+package org.languagetool.rules.ga;
+
+import org.languagetool.AnalyzedSentence;
+import org.languagetool.AnalyzedToken;
+import org.languagetool.AnalyzedTokenReadings;
+import org.languagetool.rules.*;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.ResourceBundle;
+
+import static org.languagetool.rules.ga.DhaNoBeirtData.getNumberReplacements;
+import static org.languagetool.rules.ga.DhaNoBeirtData.getDaoine;
+
+public class DhaNoBeirtRule extends Rule {
+  public DhaNoBeirtRule(ResourceBundle messages) {
+    super.setCategory(Categories.MISC.getCategory(messages));
+    setLocQualityIssueType(ITSIssueType.Misspelling);
+    addExamplePair(Example.wrong("Tá <marker>dhá</marker> dheartháireacha agam."),
+      Example.fixed("Tá <marker>beirt</marker> deartháireacha agam."));
+  }
+
+  @Override
+  public String getId() {
+    return "GA_DHA_NO_BEIRT";
+  }
+
+  @Override
+  public String getDescription() {
+    return "'dhá' nó 'beirt'";
+  }
+
+  @Override
+  public RuleMatch[] match(AnalyzedSentence sentence) {
+    List<RuleMatch> ruleMatches = new ArrayList<>();
+    AnalyzedTokenReadings[] tokens = sentence.getTokensWithoutWhitespace();
+    int markEnd = 1;
+    String replacement = null;
+    String msg = null;
+    for (int i = 1; i < tokens.length; i++) {  // ignoring token 0, i.e., SENT_START
+      if (isNumber(tokens[i]) && (i < tokens.length - 1 && isPerson(tokens[i + 1]))) {
+        markEnd = i + 1;
+        if ("dhá".equalsIgnoreCase(tokens[i].getToken())) {
+          for (int j = i + 1; j < tokens.length; j++) {
+            if ("déag".equalsIgnoreCase(tokens[j].getToken())) {
+              markEnd = j;
+              replacement = "dháréag";
+              msg = "";
+            }
+          }
+        }
+        if (replacement == null) {
+          replacement = getNumberReplacements().get(tokens[i].getToken());
+        }
+      }
+    }
+    return null;
```

Comment:
The only output of the match function is null. You probably want to include a     `return toRuleMatchArray(ruleMatches);`, but you will have to define them in line 60 and 66.

Reply:
Ah, ok - so now I know what the issue was: numbers greater than ten usually 'wrap' the noun phrase, except there's a single word for twelve as a personal number, but not as a non-personal number, which is why the lookahead for 'déag' was there.
The 'not sure what I ought to do' part was that anything that follows 'dhá' ought to be lenited, but not after 'beirt' or 'dháréag', so I was wondering if I should fix that in the suggestion (and if so, how to go about that in Java - but mostly it was the 'should' part).